src/core/raster/qgsrasterdataprovidertemporalcapabilities.h

```diff
@@ -0,0 +1,192 @@
+/***************************************************************************
+                         qgsrasterdataprovidertemporalcapabilities.h
+                         ---------------
+    begin                : February 2020
+    copyright            : (C) 2020 by Samweli Mwakisambwe
+    email                : samweli at kartoza dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSRASTERDATAPROVIDERTEMPORALCAPABILITIES_H
+#define QGSRASTERDATAPROVIDERTEMPORALCAPABILITIES_H
+
+#include "qgis_core.h"
+#include "qgis_sip.h"
+#include "qgsrange.h"
+#include "qgsdataprovidertemporalcapabilities.h"
+
+/**
+ * \class QgsRasterDataProviderTemporalCapabilities
+ * \ingroup core
+ * Implementation of data provider temporal properties for QgsRasterDataProviders.
+ *
+ * Data provider temporal capabilities reflect the temporal capabilities of a QgsDataProvider.
+ * Unlike QgsMapLayerTemporalProperties, these settings are not user-configurable,
+ * and should only be set by the QgsDataProvider itself.
+ *
+ * \since QGIS 3.14
+ */
+
+class CORE_EXPORT QgsRasterDataProviderTemporalCapabilities : public QgsDataProviderTemporalCapabilities
+{
+  public:
+
+    /**
+     * Constructor for QgsRasterDataProviderTemporalProperties.
+     */
+    QgsRasterDataProviderTemporalCapabilities( bool enabled = false );
+
+    virtual ~QgsRasterDataProviderTemporalCapabilities() = default;
+
+    /**
+     * Sets the current active datetime range for the temporal properties.
+     *
+     * \see temporalRange()
+    */
+    void setTemporalRange( const QgsDateTimeRange &dateTimeRange );
```

Comment:
I'm confused about the relationship between the temporal range, the reference temporal range, the fixed temporal range and the fixed reference temporal range.

Are these wmts specific terms? Could we expand the documentation here to clarify the role of each?

Reply:
Temporal range is the range that the current layer/dataprovider data  has,
fixed temporal range covers the lower and upper limits of the layer/data provider.
Reference temporal range represents ranges when the product/layer data where created( eg forecast data, can have forecast dates and dates when the forecast where conducted ).
Fixed reference temporal range cover the lower and upper limits for the layer reference temporal range.

Reply:
>Frequently, data are additionally temporally dependent relative to some reference time instant:
observations may have an accession time into a data repository, numerical weather forecasts may have a nominal time where observations have been assimilated to initialize the calculation, and alerts may have a time when they are issued or published. The diversity of such references precludes defining a dimension type with explicit semantics though the need for a mechanism to distinguish data based on some temporal referent is widely shared.
This document therefore defines a generic dimension called reference_time, which may be used for any of these situations.
 If it is used, however, the requirements specify that the semantics must be elucidated in the <Abstract> element of the Layer in which the dimension is defined.


from https://portal.opengeospatial.org/files/?artifact_id=56394 Section 6.2